### PR TITLE
Included Guacone Collect Github

### PR DESCRIPTION
**Pull request by nathannaveen**
# Description of the PR

* Included `./cmd/guacone collect github`. The command is very similar to `cmd/guaccollect github`.
* Also fixed the flaky test `Test_githubClient_GetWorkflow`
* Fixes https://github.com/guacsec/guac/issues/1503

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by pxp928 — APPROVED** [review]
LGTM, Not sure why the guesser is being called?

File: pkg/handler/collector/github/github.go [hunk 1]
```diff
@@ -323,6 +330,16 @@ func (g *githubCollector) fetchWorkflowRunArtifacts(ctx context.Context, docChan
 					Source:    artifact.Name,
 				},
 			}
+
+			// guess the document type and format because it is automatically set to unknown
```
[the remark this answers is not in the document]
Comment (nathannaveen) [reply pkg/handler/collector/github/github.go]: Removed it thanks!

File: cmd/guacone/cmd/github.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	Args: cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := logging.WithLogger(context.Background())
+		logger := logging.FromContext(ctx)
+
+		opts, err := validateGithubFlags(
+			viper.GetString(githubMode),
+			viper.GetString(githubSbom),
+			viper.GetString(githubWorkflowFile),
+			viper.GetString("nats-addr"),
```
Comment (jeffmendoza) [inline cmd/guacone/cmd/github.go:82]: nats addr isn't used

File: cmd/guacone/cmd/github.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+		if err := collector.Collect(cancelCtx, emit, errHandler); err != nil {
+			logger.Fatal(err)
+		}
+
+		if errFound {
+			logger.Fatalf("completed ingestion with error")
+		} else {
+			logger.Infof("completed ingestion")
+		}
```
Comment (jeffmendoza) [inline cmd/guacone/cmd/github.go:177]: Add signal handling similar to the deps dev collector:
https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/deps_dev.go#L102-L129

File: cmd/guacone/cmd/github.go (hunk 2 above)
Comment (nathannaveen) [reply cmd/guacone/cmd/github.go]: Removed

**Review by jeffmendoza — APPROVED** [review]

**Pull request merged**